### DISCUSSION TITLE
Possible to get the output from WorkerMessageHandledEvent?

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#53470] [Q&A] [answered]
I'm following the instructions to dispatch a command as a  message.

https://symfony.com/blog/new-in-symfony-6-4-more-built-in-message-handlers#run-command-handler

I have some output from that command that I'd like to see when I consume the message.  

>  In your case, you'd need to listen to the `WorkerMessageHandledEvent` and get/handle the output that way.

_Originally posted by @kbond in https://github.com/zenstruck/console-extra/issues/55#issuecomment-1874105172_

So I have the event in my listener, but how do I get OutputInterface $output?  Intuitively, it feels like there must be some way to display the output while consuming the messages.  The event only has the envelope.

```php
    #[AsEventListener(event: WorkerMessageHandledEvent::class)]
    public function onWorkerMessageHandledEvent($event): void
    {
        // get the output?
//        dd($event);
        // ...
    }
```

**Comment by MatTheCat** [comment]
`WorkerMessageHandledEvent::getEnvelope` allows to get the message’s envelope, which at this point should bear a `HandledStamp`. If the handler is `RunCommandMessageHandler`, the stamp’s `getResult` method should return a `RunCommandContext`.

**Comment by tacman** [comment]
Thanks!  I'm reliant on autocomplete but for some reason I'm not seeing how to get the stamps from the envelope.

Can this be improved?

```bash
        /** @var $event WorkerMessageHandledEvent */
        $envelope = $event->getEnvelope();
        foreach ($envelope->all() as $stamps) {
            foreach ($stamps as $stamp) {
                /** @var StampInterface $stamp */
                if ($stamp::class == HandledStamp::class) {
                    /** @var RunCommandContext $context */
                    $context = $stamp->getResult();
                    echo $context->output;
                }
            }
        }
```

**Reply by MatTheCat** [reply, marked as answer]
`Envelope::all` can take a stamp’s FQCN to only return these:

```php
/** @var $event WorkerMessageHandledEvent */
$envelope = $event->getEnvelope();
foreach ($envelope->all(HandledStamp::class) as $stamp) {
    /** @var RunCommandContext $context */
    $context = $stamp->getResult();
    echo $context->output;
}
```

Does it work?

**Reply by tacman** [reply]
Yep, thanks!  

->all() without a FQCN returns an array, which is why I iterated through it before.  Anyway, without the "s" on stamps your example works.

**Reply by MatTheCat** [reply]
> without the "s" on stamps your example works

Fixed!
